### ISSUE TITLE
x/tools/gopls: can't determine type sizes for compiler 

### ISSUE THREAD

**Issue by sunzhy** [golang/go#63751]
go  1.21
gopls v0.14.0
vscode: 1.83.1 

vscode note:
Error loading workspace: packages. Load error: can't determine type sizes for compiler "" on GOARCH ""

**Comment by dmateusp** [comment]
hit the same this morning, had to downgrade `go install golang.org/x/tools/gopls@v0.13.2`

suspecting https://go-review.googlesource.com/c/tools/+/537118 👀 

```go
        // If type size information is needed but unavailable.
	// reject the whole Load since the error is the same for every package.
	ld.sizes = types.SizesFor(response.Compiler, response.Arch)
	if ld.sizes == nil && ld.Config.Mode&(NeedTypes|NeedTypesSizes|NeedTypesInfo) != 0 {
		return nil, fmt.Errorf("can't determine type sizes for compiler %q on GOARCH %q",
			response.Compiler, response.Arch)
	}
```

**Comment by findleyr** [comment]
More questions:

Can you please report the output of the following query, run from your workspace directory?
```
go list -f "{{context.GOARCH}} {{context.Compiler}}" -- unsafe
```

Does restarting gopls@v0.14.0 fix the problem? Can you try this a few times?

**Comment by findleyr** [comment]
One more: do you by any chance have `GOPACKAGESDRIVER` set?

**Comment by dmateusp** [comment]
Version used to compile gopls: `go: go1.19.5`

go version: `go version go1.19.5 darwin/amd64`

```
❯ go list -f "{{context.GOARCH}} {{context.Compiler}}" -- unsafe
amd64 gc
```

No amount of restarting seems to be doing the trick

Yes, my `GOPACKAGESDRIVER` is set through this command: `exec bazel run -- @io_bazel_rules_go//go/tools/gopackagesdriver "${@}"`

Note: I'm using Bazel and VSCode, all the go commands I ran from `.dist/mypkg/external/go_sdk/bin/go`

**Comment by findleyr** [comment]
Awesome, thank you for following up. In that case, I think we understand this and will cut a patched prerelease soon. In short: the `Arch` and `Compiler` field were added to the driver response, and `go/packages` was not tolerant of missing fields.

It looks like the bazel driver was updated to set the missing fields in https://github.com/bazelbuild/rules_go/commit/6e10f8c264e3577b0e2fb16cfd2d37cff05cdcfa, so you may also be able to fix this by updating bazel.

For other folks paying attention to this issue: can anyone else please confirm that they have a `GOPACKAGESDRIVER` set? We believe that is the only way to hit this bug.

**Comment by findleyr** [comment]
This should be fixed in v0.14.1-pre.1:

```
go install golang.org/x/tools/gopls@v0.14.1-pre.1
```

Could anyone here please try that out to confirm. With your confirmation, we will release v0.14.1 tomorrow.

**Comment by findleyr** [comment]
Since it's Friday and I have yet to hear confirmation of the fix, let's defer the release until Monday.

**Comment by Urvik08** [comment]
> ```
> go install golang.org/x/tools/gopls@v0.14.1-pre.1
> ```

This worked for me, thank you!

**Comment by findleyr** [comment]
@Urvik08 thanks for confirming!

**Comment by seanmorton-afs** [comment]
@findleyr `v0.14.1-pre.1` also fixed the problem for me, and fyi yes I'm also using `GOPACKAGESDRIVER=exec bazel run -- @io_bazel_rules_go//go/tools/gopackagesdriver "${@}"` 

Thank you for the fix! Have a great weekend.

**Comment by Minion-Zhang** [comment]
> ```
> go install golang.org/x/tools/gopls@v0.14.1-pre.1
> ```

This works for me too. I'm also using Bazel and `go version go1.21.3 linux/amd64`. Thanks for the quick fix!

**Comment by sunzhy** [comment]
This works for me too. I'm also using Bazel. Thanks for the quick fix!

**Comment by findleyr** [comment]
Thanks all. This is released as gopls@v0.14.1:

```
go install golang.org/x/tools/gopls@v0.14.1
```
